platform/darwin/test/MGLSourceQueryTests.m

```diff
@@ -0,0 +1,73 @@
+#import <Mapbox/Mapbox.h>
+
+#import "NSBundle+MGLAdditions.h"
+
+#import <XCTest/XCTest.h>
+#if TARGET_OS_IPHONE
+    #import <UIKit/UIKit.h>
+#else
+    #import <Cocoa/Cocoa.h>
+#endif
+
+@interface MGLSourceQueryTests : XCTestCase <MGLMapViewDelegate>
+
+@property (nonatomic) MGLMapView *mapView;
+@property (nonatomic) MGLStyle *style;
+
+@end
+
+@implementation MGLSourceQueryTests {
+    XCTestExpectation *_styleLoadingExpectation;
+}
+
+- (void)setUp {
+    [super setUp];
+    
+    [MGLAccountManager setAccessToken:@"pk.feedcafedeadbeefbadebede"];
+    NSURL *styleURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"query-style" withExtension:@"json"];
+    self.mapView = [[MGLMapView alloc] initWithFrame:CGRectMake(0, 0, 100, 100) styleURL:styleURL];
+    self.mapView.delegate = self;
+    if (!self.mapView.style) {
+        _styleLoadingExpectation = [self expectationWithDescription:@"Map view should finish loading style."];
+        [self waitForExpectationsWithTimeout:1 handler:nil];
+    }
+}
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    XCTAssertNotNil(mapView.style);
+    XCTAssertEqual(mapView.style, style);
+    
+    [_styleLoadingExpectation fulfill];
```

Comment:
In case you’re wondering why this works for shape sources but not vector sources, `-mapView:didFinishLoadingStyle:` gets called as soon as we’re done loading and parsing the style itself. It isn’t guaranteed that all sources load by this point. If you wanted to query a vector source too, you’d need to wait until `-mapViewDidFinishLoadingMap:` is called. But this is perfectly fine for the inline GeoJSON sources in query-style.json, since they load as part of the style.